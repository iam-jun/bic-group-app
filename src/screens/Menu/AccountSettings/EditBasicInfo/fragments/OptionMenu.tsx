import React from 'react';
import {StyleSheet, View, TouchableOpacity} from 'react-native';
import {ExtendedTheme, useTheme} from '@react-navigation/native';

import {IOptionItem} from '~/interfaces/IEditUser';

import BottomSheet from '~/beinComponents/BottomSheet';
import Divider from '~/beinComponents/Divider';
import Text from '~/beinComponents/Text';
import PrimaryItem from '~/beinComponents/list/items/PrimaryItem';
import i18next from 'i18next';
import Icon from '~/beinComponents/Icon';
import spacing from '~/theme/spacing';

interface OptionMenuProps {
  data: IOptionItem[];
  menuRef: any;
  value: string;
  title: string;
  onItemPress: (item: any) => void;
  testID?: string;
}

const OptionMenu = ({
  data,
  value,
  title,
  menuRef,
  onItemPress,
  testID,
}: OptionMenuProps) => {
  const theme = useTheme() as ExtendedTheme;

  const renderItem = ({item}: {item: IOptionItem}) => {
    return (
      <TouchableOpacity
        testID={`edit_user_info.option_menu.item_${item.type}`}
        onPress={() => onItemPress(item)}>
        <PrimaryItem
          title={i18next.t(item.title)}
          height={36}
          RightComponent={
            value === item.type ? (
              <Icon
                icon={'Check'}
                size={24}
                tintColor={theme.colors.purple60}
              />
            ) : undefined
          }
        />
      </TouchableOpacity>
    );
  };

  return (
    <View testID={testID}>
      <BottomSheet
        modalizeRef={menuRef}
        ContentComponent={
          <View style={styles.contentComponent}>
            <Text.ButtonSmall
              color={theme.colors.gray50}
              style={styles.chooseText}
              useI18n>
              {title}
            </Text.ButtonSmall>
            <Divider />
            {data.map((item: IOptionItem) => (
              <View key={item?.title + item?.type}>{renderItem({item})}</View>
            ))}
          </View>
        }
      />
    </View>
  );
};

export default OptionMenu;

const styles = StyleSheet.create({
  contentComponent: {},
  chooseText: {
    margin: spacing.margin.base,
  },
});
