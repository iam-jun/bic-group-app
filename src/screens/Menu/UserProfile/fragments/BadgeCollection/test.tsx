import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
