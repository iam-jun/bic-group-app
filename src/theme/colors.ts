export default {
  dark: {
    colors: {
      transparent: 'rgba(255,255,255,0)',
      primary: '#5D2C9A',
      primary1: '#F8F5FF',
      primary2: '#EADFFE',
      primary3: '#DBCAFD',
      primary4: '#BC9FF8',
      primary5: '#A17AF2',
      primary6: '#7843DD',
      primary7: '#421187',

      background: '#20272F',
      bgSecondary: '#2F3945',
      bgFocus: '#505E71',
      bgHover: '#3F4B5B',
      bgDisable: '#505E71',
      bgButtonPrimary: '#8B5CE9',
      bgButtonSecondary: '#20272E',
      BorderButtonSecondary: '#8B5CE9',
      borderFocus: '#627286',
      borderDisable: '#627286',
      borderCard: '#B2BDCD',
      borderDivider: '#3F4B5B',
      textInput: '#F9FAFB',
      placeholder: '#3F4B5B',
      surface: '#2F3945', //Neutral1 background color for elements containing content, such as cards.
      iconTint: '#D5DCE5',
      iconTintLight: '#77879C',
      iconTintReversed: '#2F3945',

      link: '#2396FF',
      activeLink: '#5B22B8',
      success: '#0AC458',
      error: '#E8455F',
      warning: '#FFBC1F',
      statusOrange: '#E98A32',

      textPrimary: '#F9FAFB',
      textSecondary: '#77879C',
      textSuccess: '#FFFFFF',
      textDanger: '#FFFFFF',
      textWarning: '#2F3945',
      textInfo: '#007AFF',
      textDisabled: '#627286',
      textReversed: '#2F3945',
      textTertiary: '#627286',

      facebook: '#2D88FF',
      google: '#FF2828',
      apple: '#FFFFFF',

      //react-native-paper default field
      accent: '#6600cc', //secondary color for your app which complements the primary color.
      text: '#29272A', //text color for content.
      disabled: 'rgba(0, 0, 0, 0.26)', //color for disabled elements.
      backdrop: 'rgba(0, 0, 0, 0.5)', //color for backdrops of various components such as modals.
      onSurface: '#000000', //background color for snackbars
      notification: '#f50057', //background color for badges

      //TabBar
      tabBarBackground: '#2F3945', //Neutral0
      activeColor: '#8B5CE9', //Primary50
      inactiveColor: '#77879C', //Neutral50

      // ViewLabel
      ViewLabel: {
        color: '#999999',
        colorHeading: '#000000',
      },
    },
  },
  light: {
    colors: {
      transparent: 'rgba(255,255,255,0)',
      primary: '#5D2C9A', //primary is brand color
      primary1: '#F8F5FF',
      primary2: '#EADFFE',
      primary3: '#DBCAFD',
      primary4: '#BC9FF8',
      primary5: '#A17AF2',
      primary6: '#7843DD', //Main Brand
      primary7: '#421187', //Primary90

      background: '#FFFFFF', //Neutral0
      bgSecondary: '#F9FAFB', //Neutral1
      bgFocus: '#D5DCE5', //Neutral10
      bgHover: '#EAEDF2', //Neutral5
      bgDisable: '#D5DCE5', //Neutral10
      bgButtonPrimary: '#7843DD ', //Primary50
      bgButtonSecondary: '#FFFFFF', //Neutral0'
      BorderButtonSecondary: '#7843DD',
      borderFocus: '#B2BDCD', //Neutral20
      borderDisable: '#B2BDCD', //Neutral20
      borderCard: '#B2BDCD', //Neutral20
      borderDivider: '#EAEDF2', //`Neutral5
      textInput: '#2F3945', // Neutral80
      placeholder: '#EAEDF2', //Neutral5
      surface: '#F9FAFB', //Neutral1 background color for elements containing content, such as cards.
      iconTint: '#2F3945', //Neutral80
      iconTintLight: '#627286', //Neutral50
      iconTintReversed: '#FFFFFF', //Neutral0

      link: '#007AFF', //Blue50
      activeLink: '#421187', //Primary90
      success: '#00A64D', //Green50
      error: '#DD2C3E', //Red50
      warning: '#FFBC1F', //Yellow40
      statusOrange: '#E98A32', //Orange40

      textPrimary: '#2F3945', //Neutral80
      textSecondary: '#627286', //Neutral50
      textSuccess: '#FFFFFF', //Neutral0
      textDanger: '#FFFFFF', //Neutral0
      textWarning: '#2F3945', //Neutral80
      textInfo: '#007AFF', //Blue50
      textDisabled: '#B2BDCD', //Neutral20
      textReversed: '#FFFFFF', //Neutral0
      textTertiary: '#627286',

      facebook: '#2D88FF',
      google: '#FF2828',
      apple: '#FFFFFF',

      //react-native-paper default field
      accent: '#6600cc', //secondary color for your app which complements the primary color.
      text: '#29272A', //text color for content.
      disabled: 'rgba(0, 0, 0, 0.26)', //color for disabled elements.
      backdrop: 'rgba(0, 0, 0, 0.5)', //color for backdrops of various components such as modals.
      onSurface: '#000000', //background color for snackbars
      notification: '#f50057', //background color for badges

      //TabBar
      tabBarBackground: '#FFFFFF', //Neutral0
      activeColor: '#7843DD', //Primary50
      inactiveColor: '#627286', //Neutral50

      // ViewLabel
      ViewLabel: {
        color: '#999999',
        colorHeading: '#000000',
      },
    },
  },
};
