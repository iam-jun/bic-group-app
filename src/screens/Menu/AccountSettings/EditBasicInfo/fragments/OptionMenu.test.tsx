import React from 'react';

import {fireEvent} from '~/test/testUtils';
import {render, cleanup} from '@testing-library/react-native';
import colors from '~/theme/theme';

import OptionMenu from './OptionMenu';

afterEach(cleanup);

describe('OptionMenu conponent', () => {
  const baseProps = {
    data: [
      {title: 'settings:text_male', type: 'MALE'},
      {title: 'settings:text_female', type: 'FEMALE'},
      {title: 'settings:text_others', type: 'OTHERS'},
    ],
    value: 'FEMALE',
    menuRef: null,
    title: 'settings:title_choose_gender',
  };

  it('renders correctly', () => {
    const onGenderItemPress = jest.fn();

    const rendered = render(
      <OptionMenu {...baseProps} onItemPress={onGenderItemPress} />,
    ).toJSON();

    expect(rendered).toMatchSnapshot();
  });

  it('should call props onItemPress', () => {
    const onGenderItemPress = jest.fn();

    const rendered = render(
      <OptionMenu {...baseProps} onItemPress={onGenderItemPress} />,
    );

    const btnOpenBottomSheetCom = rendered.getByTestId(
      'edit_user_info.option_menu.item_MALE',
    );
    expect(btnOpenBottomSheetCom).toBeDefined();
    fireEvent.press(btnOpenBottomSheetCom);

    expect(onGenderItemPress).toHaveBeenCalledWith({
      title: 'settings:text_male',
      type: 'MALE',
    });
  });
});
