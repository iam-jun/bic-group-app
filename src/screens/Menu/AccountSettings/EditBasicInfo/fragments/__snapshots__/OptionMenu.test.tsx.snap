// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionMenu conponent renders correctly 1`] = `
<View>
  <View>
    <View
      style={Object {}}
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2F3945",
              "fontFamily": "OpenSans-SemiBold",
              "fontSize": 12,
            },
            Object {
              "color": "#627286",
            },
            Object {
              "margin": 12,
            },
          ]
        }
      >
        Choose your gender
      </Text>
      <View
        style={
          Array [
            Object {
              "alignSelf": "stretch",
              "backgroundColor": "#EAEDF2",
              "height": 1,
              "marginHorizontal": 0,
              "width": undefined,
            },
            Object {},
          ]
        }
        testID="divider"
      />
      <View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="edit_user_info.option_menu.item_MALE"
        >
          <View
            className=""
          >
            <View
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={false}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 36,
                  "opacity": 1,
                  "paddingHorizontal": 12,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  numberOfLines={2}
                  style={
                    Array [
                      Object {
                        "color": "#2F3945",
                        "fontFamily": "OpenSans-SemiBold",
                        "fontSize": 14,
                      },
                      Object {},
                      Object {},
                    ]
                  }
                >
                  Male
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="edit_user_info.option_menu.item_FEMALE"
        >
          <View
            className=""
          >
            <View
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={false}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 36,
                  "opacity": 1,
                  "paddingHorizontal": 12,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  numberOfLines={2}
                  style={
                    Array [
                      Object {
                        "color": "#2F3945",
                        "fontFamily": "OpenSans-SemiBold",
                        "fontSize": 14,
                      },
                      Object {},
                      Object {},
                    ]
                  }
                >
                  Female
                </Text>
              </View>
              <View
                accessibilityState={
                  Object {
                    "disabled": true,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={false}
                hitSlop={
                  Object {
                    "bottom": 10,
                    "left": 10,
                    "right": 10,
                    "top": 10,
                  }
                }
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      },
                      undefined,
                      Object {
                        "backgroundColor": undefined,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        undefined,
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    <View
                      className="icon-wrapper"
                    >
                      <RNSVGSvgView
                        align="xMidYMid"
                        bbHeight={24}
                        bbWidth={24}
                        fill="#421187"
                        focusable={false}
                        height={24}
                        meetOrSlice={0}
                        minX={0}
                        minY={0}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "borderWidth": 0,
                            },
                            Object {},
                            Object {
                              "flex": 0,
                              "height": 24,
                              "width": 24,
                            },
                          ]
                        }
                        vbHeight={24}
                        vbWidth={24}
                        width={24}
                      >
                        <RNSVGGroup
                          fill={4282519943}
                          propList={
                            Array [
                              "fill",
                            ]
                          }
                        >
                          <RNSVGPath
                            d="M18.71,7.21a1,1,0,0,0-1.42,0L9.84,14.67,6.71,11.53A1,1,0,1,0,5.29,13l3.84,3.84a1,1,0,0,0,1.42,0l8.16-8.16A1,1,0,0,0,18.71,7.21Z"
                          />
                        </RNSVGGroup>
                      </RNSVGSvgView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="edit_user_info.option_menu.item_OTHERS"
        >
          <View
            className=""
          >
            <View
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={false}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 36,
                  "opacity": 1,
                  "paddingHorizontal": 12,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  numberOfLines={2}
                  style={
                    Array [
                      Object {
                        "color": "#2F3945",
                        "fontFamily": "OpenSans-SemiBold",
                        "fontSize": 14,
                      },
                      Object {},
                      Object {},
                    ]
                  }
                >
                  Others
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
